### PR TITLE
Remove -Werror

### PR DESCRIPTION
**Pull request by Hinara**
-Werror should never be use in build as this might prompt new errors when a compiler choose to add some new warning by default.

### CODE REVIEW

**Comment by 123jimin** [comment]
*For developing USC*, turning on `-Werror` is usually better since it often catches coding mistakes while compiling. One problem while developing USC is that for `msvc` the related flag `/WX` is not set and the set of warnings vary quite a lot between `msvc` and `gcc`, which sometimes causes actions failing despite of successful local builds. But I'm willing to live with that and it can be improved by adjusting flags for `msvc`.

However, *for just building USC from source*, `-Werror` would indeed be problematic because previously-succeeded build might be failed depending on compiler version, as you said. For this case, adding an option for disabling `-Werror` ([an example](https://github.com/json-c/json-c/issues/489)) would be beneficial.

TL;DR: I don't like removing `-Werror`, but how about adding a flag for disabling `-Werror` for cmake?

**Comment by Hinara** [comment]
I think it should be the invert !
CMake by default should not put the flag, but if would be enables if you use a some defines.
And if you want to enable a flag that you want just modify your CMakeCache and add it.
Why in this way ?
Because end user should not have to know the define of your project to make it run by default.
Secondly `-Werror` does not 
>  it often catches coding mistakes while compiling
It just convert warning to errors. So as long as you don't ignore warnings everything should be fine.

**Comment by Hinara** [comment]
So there is already a flag in cmake to do what you want `-Werror=dev`

**Comment by Drewol** [comment]
-Werror is now off by default but still on in the CI which should catch most things
